### PR TITLE
Compliation errors cause the build to fail now, dojo/cli-test-itern#31

### PR DESCRIPTION
**Pull request by rorticus**
**Type:** bug

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Errors during the webpack build will now return a rejected promise. With this, `dojo test` will fail when you have build errors.

Resolves https://github.com/dojo/cli-test-intern/issues/31

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -196,6 +197,14 @@ function compile(config: webpack.Config, options: WebpackOptions): Promise<void>
 				}
 
 				console.log(stats.toString(options.stats));
+
+				if (stats.compilation && stats.compilation.errors && stats.compilation.errors.length > 0) {
+					reject({
+						exitCode: 1,
+						message: ''
```
Comment (kitsonk) [inline src/main.ts]: surely we can provide something more informative?

Comment (rorticus) [reply src/main.ts]: Well.. I tried something more informative but it just looked silly. At this point the webpack build has already printed out all the errors so the console is full of the _real_ error details.  I had could try something like `The build failed with errors` or something, but like I said, it seemed obvious from the red text above it... @kitsonk thoughts?

Comment (kitsonk) [reply src/main.ts]: Hmm... slightly raises the question of if we have a `--force` or not... which then this message could indicate that there are compilation errors reported, use `--force` to ignore.

Comment (rorticus) [reply src/main.ts]: I like that. I will add support for `--force`! Can I print out `use the --force to continue anyways`? 😛

Comment (kitsonk) [reply src/main.ts]: These aren't the 🤖 you are looking for.

**Comment by mwistrand** [comment]
This also addresses at least part of #207, correct?

**Comment by rorticus** [comment]
Good catch @mwistrand , yes!

**Pull request merged**